### PR TITLE
Feature/122 new barrister fields

### PR DESCRIPTION
**Pull request by tomlovesgithub**
Will need to be some wording changes most likely and theres a slight bug, which ive noted with a @FIXME@ concerning the date Input not being cleared/overwritten when 'did not complete pupilage' is selected...also i accidentally named the branch after the issue being blocked by this work, not the issue itself 🤦

### CODE REVIEW

File: src/components/Form/DateInput.vue [hunk 1]
```diff
@@ -117,6 +117,9 @@ export default {
       year: null,
     };
   },
+  mounted(){
+    console.log(this.value);
```
Comment (crablab) [inline src/components/Form/DateInput.vue:121]: I assume this is for debugging? :-) 

`eslint` should not be passing with `console` statements :thinking:

Comment (tomlovesgithub) [reply src/components/Form/DateInput.vue:121]: yeah, tbh thought i'd removed this! thanks for catching it, think i bodged a rebase and smooshed this back into the commit

Comment (crablab) [reply src/components/Form/DateInput.vue]: Weird that it didn't get caught in linting though! :upside_down_face:  Oh well...

File: src/views/Apply/FinalCheck/Review.vue [hunk 2]
```diff
@@ -771,7 +771,26 @@
                 </dd>
               </div>
 
-              <div class="govuk-summary-list__row">
+              <div 
+                v-if="item.type==='barrister' && item.date"
```
Comment (YaaL) [inline src/views/Apply/FinalCheck/Review.vue:775]: You can split the condition to make it more readable: keep `v-if="item.date"` on the row div and add a `<template v-if="item.type === 'barrister'">` inside.

File: src/components/RepeatableFields/Qualification.vue [hunk 3]
```diff
@@ -43,8 +43,36 @@
...
+      v-if="row.type==='barrister'"
+    >
+      <DateInput
+        :id="qualificationDate"
+        :v-model="row.date"
+        :value="row.date ? row.date : null"
+        label="When did you complete pupillage?"
+        type="month"
+      />
+      <CheckboxGroup
```
Comment (YaaL) [inline src/components/RepeatableFields/Qualification.vue:56]: We only have yes/no, so it should be radio not checkbox

Comment (YaaL) [inline src/components/RepeatableFields/Qualification.vue:56]: You don't need to use `CheckboxGroup` here, we have a single `Checkbox` component for situations like this

File: src/components/RepeatableFields/Qualification.vue [hunk 4]
```diff
@@ -43,29 +43,43 @@
...
+      :label="row.type==='barrister'?'When did you complete pupillage?':'When did you qualify?'"
       type="month"
     />
+    <Checkbox 
+      :id="qualificationNotComplete"
+      v-model="row.qualificationNotComplete"
+      label="I did not complete pupillage"
+    />
+    <TextareaInput
+      v-if="row.completedPupillage && row.completedPupillage[0] === false"
```
Comment (warrensearle) [inline src/components/RepeatableFields/Qualification.vue:58]: This doesn't look right. Should it be `v-if="row.qualificationNotComplete"`?

File: src/views/Apply/FinalCheck/Review.vue [hunk 5]
```diff
@@ -771,9 +771,12 @@
...
               </div>
 
-              <div class="govuk-summary-list__row">
+              <div 
+                v-if="item.date"
+                class="govuk-summary-list__row"
+              >
                 <dt class="govuk-summary-list__key">
-                  Date qualified
+                  {{ item.type === 'barrister' ? "Date completed pupillage" : "Date qualified" }}
```
Comment (warrensearle) [inline src/views/Apply/FinalCheck/Review.vue:779]: You could use `v-if` here instead, like:
```
<dt v-if="item.type === 'barrister'" class="govuk-summary-list__key">Date completed pupillage</dt>
<dt v-else class="govuk-summary-list__key">Date qualified</dt>
```

File: src/components/RepeatableFields/Qualification.vue (hunk 4 above)
Comment (tomlovesgithub) [reply src/components/RepeatableFields/Qualification.vue:58]: yeah i think i may have rebased incorrectly, will sort this now

File: src/views/Apply/FinalCheck/Review.vue (hunk 5 above)
Comment (tomlovesgithub) [reply src/views/Apply/FinalCheck/Review.vue:779]: ✅

**Pull request merged**